modules/drivers/mongo/test/metabase/driver/mongo_test.clj

```diff
@@ -46,40 +46,42 @@
 (deftest can-connect-test?
   (mt/test-driver
    :mongo
-   (doseq [{:keys [details expected message]} [{:details  {:host   "localhost"
-                                                           :port   3000
-                                                           :dbname "bad-db-name"}
-                                                :expected false}
-                                               {:details  {}
-                                                :expected false}
-                                               {:details  {:host   "localhost"
-                                                           :port   27017
-                                                           :user   "metabase"
-                                                           :pass   "metasample123"
-                                                           :dbname "metabase-test"}
-                                                :expected true}
-                                               {:details  {:host   "localhost"
-                                                           :user   "metabase"
-                                                           :pass   "metasample123"
-                                                           :dbname "metabase-test"}
-                                                :expected true
-                                                :message  "should use default port 27017 if not specified"}
-                                               {:details  {:host   "123.4.5.6"
-                                                           :dbname "bad-db-name?connectTimeoutMS=50"}
-                                                :expected false}
-                                               {:details  {:host   "localhost"
-                                                           :port   3000
-                                                           :dbname "bad-db-name?connectTimeoutMS=50"}
-                                                :expected false}
-                                               {:details  {:conn-uri "mongodb://metabase:metasample123@localhost:27017/metabase-test?authSource=admin"}
-                                                :expected (not (tdm/ssl-required?))}
-                                               {:details  {:conn-uri "mongodb://localhost:3000/bad-db-name?connectTimeoutMS=50"}
-                                                :expected false}]
-           :let [ssl-details (tdm/conn-details details)]]
-     (testing (str "connect with " details)
-       (is (= expected
-              (driver.u/can-connect-with-details? :mongo ssl-details))
-           (str message))))))
+   (mt/dataset sample-dataset
+     (mt/db)
```

Comment:
This test was effectively broken. It should have been failing but just happened to work. `dbname` was actually never checked to see if it was there or not, now it is checked, and I've updated the test to use the sample-dataset.